### DISCUSSION TITLE
Rules: Which rules is used to when a new record is created

### DISCUSSION THREAD

**Discussion by teezyfortune** [codatio/codat-docs#298] [Q&A] [answered]
Which rule type should I listen when creating  the following entities(customer, company), etc is created

**Comment by mcclowes** [comment]
Hi @teezyfortune,

We don't have a rule for just the creation of a Company. If you're creating the company via API, you'll get a response to show it's been created successfully, and if you're using Link, you usually want to wait until the connection is established or data has been synchronized. The rules dfor these are here:
- https://docs.codat.io/introduction/webhooks/core-rules-types#company-data-connection-status-changed
- https://docs.codat.io/introduction/webhooks/core-rules-types#new-company-synchronized

**Reply by teezyfortune** [reply]
Got it @mcclowes , what about the customer and suppliers? Also, I got a validation error whenever I tried to push a bill, that I am missing some fields, but I have all fields in the object, I don't know why I am getting that error.

**Reply by mcclowes** [reply, marked as answer]
You can also listen for our 'data set changed' hook (https://docs.codat.io/introduction/webhooks/core-rules-types), and check for Data.dataType being the data type you're interested in.

Re: creating bills:
- Are you able to share what you're pushing, which integration it is, and what you get back?
- Are you calling this endpoint first? https://docs.codat.io/accounting-api#/operations/get-create-update-bills-model

The fields required can change a bit based on the integration. You can see examples of the required fields in this dropdown on that page:


![Screenshot 2023-04-06 at 10 00 27](https://user-images.githubusercontent.com/4573407/230329644-3470a8d0-d4db-4719-af2f-0a144354aaa3.png)

**Reply by teezyfortune** [reply]
Thank you. I will check that
